Source/WebCore/dom/make-event-names.py

```diff
@@ -0,0 +1,204 @@
+#!/usr/bin/env python3
+#
+# Copyright (C) 2024 Apple Inc. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+# 1. Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+# 2. Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+# THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+# BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+# THE POSSIBILITY OF SUCH DAMAGE.
+
+import argparse
+import json
+
+
+def lowercaseFirstLetter(string):
+    return string[0].lower() + string[1:] if len(string) and string[1].islower() else string
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Generate EventNames.h')
+    parser.add_argument('--event-names')
+    args = parser.parse_args()
+
+    with open(args.event_names, "r", encoding="utf-8") as event_names_file:
+        event_names_input = json.load(event_names_file)
+
+    with open('EventNames.h', 'w') as output_file:
+        def write(text):
+            output_file.write(text)
+
+        def writeln(text):
+            write(text + "\n")
```

Comment:
If we can somehow share the writer utilities from CSSPseudoSelectors.json & CSSProperties.json (they're duplicated in both of those), you might be able to reuse those if you want to.

Reply:
Hm... that would require quite a bit of refactoring so kept as is for now.